azkaban-common/src/main/java/azkaban/scheduler/Schedule.java

```diff
@@ -273,26 +244,20 @@ private DateTime getNextRuntime(long scheduleTime, DateTimeZone timezone,
     return date;
   }
 
+  /**
+   *
+   * @param scheduleTime represents the time when Schedule Servlet receives the Cron Schedule API calling.
+   * @param timezone is always UTC (after 3.1.0)
+   * @param ce
+   * @return the First Scheduled DateTime to run this flow.
+   */
   private DateTime getNextCronRuntime(long scheduleTime, DateTimeZone timezone,
       CronExpression ce) {
-    Date now = new DateTime().toDate();
-    Date date = new DateTime(scheduleTime).withZone(timezone).toDate();
-    int count = 0;
-    while (now.compareTo(date) >= 0) {
-      if (count > 100000) {
-        throw new IllegalStateException(
-            "100000 increments of period did not get to present time.");
-      }
-
-      if (ce == null) {
-        break;
-      } else {
-        date = ce.getNextValidTimeAfter(date);
-      }
 
-      count += 1;
+    Date date = new DateTime(scheduleTime).withZone(timezone).toDate();
```

Comment:
This method will no longer take into consideration of the current time. Is it right?